File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        ls = []
        for i in range(len(nums)):
            if target-nums[i] in hash:
                ls.append(hash[target-nums[i]])
                ls.append(i)
                return ls
            else:
                hash[nums[i]] = i
        return ls

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    dict := map[int]int{} 
    for i,v := range nums{
        t := target - v
        ind,exists := dict[t]
        if exists {
            return []int{ind,i}
        }
        dict[v] = i
    }
    return []int{}
}